### PR TITLE
Add WebSocket support for Play

### PR DESCRIPTION
**Pull request by ghostdogpr**
I'm pretty new to tapir, hopefully it makes sense 😄

### CODE REVIEW

File: server/play-server/src/main/scala/sttp/tapir/server/play/PlayServerInterpreter.scala [hunk 1]
```diff
@@ -50,46 +52,69 @@ trait PlayServerInterpreter {
...
         }
       }
     }
   }
 
+  private def isWebSocket(header: RequestHeader): Boolean =
+    (for {
+      connection <- header.headers.get(sttp.model.HeaderNames.Connection)
+      upgrade <- header.headers.get(sttp.model.HeaderNames.Upgrade)
+    } yield connection.equalsIgnoreCase("Upgrade") && upgrade.equalsIgnoreCase("websocket")).getOrElse(false)
```
Comment (paulpdaniels) [inline server/play-server/src/main/scala/sttp/tapir/server/play/PlayServerInterpreter.scala:116]: `Upgrade` will contain the subprotocols which may not be `websocket` so I think we just need to check for the presence of this header.

Comment (ghostdogpr) [reply server/play-server/src/main/scala/sttp/tapir/server/play/PlayServerInterpreter.scala:116]: I don't think so? I just checked and the protocol is passed in another header called `sec-websocket-protocol`. 

Looking at the spec: 
> If the response lacks an |Upgrade| header field or the |Upgrade|
       header field contains a value that is not an ASCII case-
       insensitive match for the value "websocket", the client MUST
       _Fail the WebSocket Connection_.

Actually I got this logic from here: https://github.com/softwaremill/tapir/blob/41c6bc7ca656869eb9c7eed7f4f850364968c9bd/core/src/main/scala/sttp/tapir/Tapir.scala#L368

Comment (paulpdaniels) [reply server/play-server/src/main/scala/sttp/tapir/server/play/PlayServerInterpreter.scala:116]: Ah you are right I was just looking at the top part of this document: https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism#upgrading_to_a_websocket_connection

**Comment by adamw** [comment]
Yes looks good, thanks :)

**Pull request merged**